openchrom/plugins/net.openchrom.nmr.processing.supplier.base/src/net/openchrom/nmr/processing/supplier/base/core/IcoShiftAlignment.java

```
@@ -429,8 +420,17 @@ private SimpleMatrix extractMultipleSpectra(Collection<? extends IMeasurementNMR
 		SimpleMatrix experimentalDatasetsMatrixPartForProcessing = extractPartOfDataForProcessing(experimentalDatasetsMatrix, referenceWindow);
 		//
 		IcoShiftAlignmentTargetCalculationSelection targetCalculationSelection = settings.getTargetCalculationSelection();
-		if((targetCalculationSelection == IcoShiftAlignmentTargetCalculationSelection.MAX) || (targetCalculationSelection == IcoShiftAlignmentTargetCalculationSelection.CALIBRATE)) {
-			// calculate max target here for each interval; "CALIBRATE" applies here too
+		if(calculateCalibrationTargetFunction != null) {
+			// calculate target used for calibration here
+			targetForProcessing = calculateCalibrationTargetFunction.calculateCalibrationTarget(experimentalDatasetsMatrix, interval, settings);
+			if(targetForProcessing == null) {
```

you better throw this exception from calculateCalibrationTarget instead of returning null. And you should describe better what the cause if for 'could not be calculated correctly'

---

This check - introduced by Jan - does not make any sense. The calculation depends mainly on the size of the interval and will always return an targetForProcessing != null. I added an exception that will be thrown if inside the target calculation one of the parameters would be set to less than zero.